### PR TITLE
[[ Foundation ]] CFTypeRef Bridges

### PR DESCRIPTION
**Pull request by runrevmark**
This patch cleans up and completes the set of CoreFoudation <->
Foundation type bridging functions in libfoundation and exposes
them to LCB through the objc module.

### CODE REVIEW

**Comment by livecodeali** [comment]
You need to change the source file that's removed here: https://github.com/livecode/livecode/pull/6160/files#diff-e0a1c562b02c099363ab3b194b71bf6bR172 to foundation-cf.cpp

File: libfoundation/src/foundation-cf.cpp [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+{
+    MCAutoArrayRef t_array;
+    if (!MCArrayCreateMutable(&t_array))
+    {
+        return false;
+    }
+    
+    CFDictionaryApplyFunction(p_cf_dict, nullptr, nullptr);
+    
+    return false;
```
Comment (livecodeali) [inline libfoundation/src/foundation-cf.cpp]: This should probably throw some kind of unimplemented error, if it's meant to be unimplemented. Or at least a comment :-)

Comment (runrevmark) [reply libfoundation/src/foundation-cf.cpp]: Heh - or actually implement it :)

**Comment by livecodeali** [comment]
@livecode-vulcan review ok f798356

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok f798356bf3a8cb1e5d4d387e7a38aa6fab348d91

**Comment by livecode-vulcan** [comment]
😞 test failure f798356bf3a8cb1e5d4d387e7a38aa6fab348d91

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1004)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1020)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/994)
* try-community-universal-ios-iphoneos11.1: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.1/builds/67)
* try-community-universal-ios-iphonesimulator11.1: [failure](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.1/builds/67)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/999)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1021)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1004)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1020)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1003)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1013)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1036)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1000)

**Comment by livecodeali** [comment]
@livecode-vulcan review ok f467757

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok f467757e4986bf98c1833b3640c267088560bd36

**Comment by livecode-vulcan** [comment]
😎 test success f467757e4986bf98c1833b3640c267088560bd36

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1007)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1023)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/997)
* try-community-universal-ios-iphoneos11.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.1/builds/70)
* try-community-universal-ios-iphonesimulator11.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.1/builds/70)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1002)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1024)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1007)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1023)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1006)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1016)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1039)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1003)

**Pull request merged**